Extension/swhttprequest.cpp

```diff
@@ -511,6 +511,58 @@ static cell_t sm_GetHTTPResponseBodyCallback(IPluginContext *pContext, const cel
 	return 1;
 }
 
+static cell_t sm_SetHTTPRequestRawPostBodyFromFile(IPluginContext *pContext, const cell_t *params)
+{
+	ISteamHTTP *pHTTP;
+	SteamWorksHTTPRequest *pRequest = GetRequestPointer(pHTTP, pContext, params[1]);
+
+	if (pRequest == NULL)
+	{
+		return 0;
+	}
+
+	char *pContentType;
+	pContext->LocalToString(params[2], &pContentType);
+
+	char *pFilePath;
+	pContext->LocalToString(params[3], &pFilePath);
+
+	char szFinalPath[PLATFORM_MAX_PATH];
+	smutils->BuildPath(Path_Game, szFinalPath, sizeof(szFinalPath), "%s", pFilePath);
+
+	FILE *pInputFile = fopen(szFinalPath, "rb");
+
+	if (!pInputFile)
+	{
+		return pContext->ThrowNativeError("Unable to open %s for reading. errno: %d", szFinalPath, errno);
+	}
+
+	uint32_t size;
+
+	fseek(pInputFile, 0, SEEK_END);
+	size = ftell(pInputFile);
+	fseek(pInputFile, 0, SEEK_SET);
+
+	if (size <= 0)
+	{
+		fclose(pInputFile);
+		return 0;
+	}
+
+	char *pBuffer = new char[size + 1];
+	fread(pBuffer, sizeof(char), size, pInputFile);
```

Comment:
not sure if this is ignored otherwise in the code add, but fread’s retval should probably be checked to ensure a read didn’t fail.